src/lib/y2users/linux/useradd_config_writer.rb

```diff
@@ -0,0 +1,174 @@
+      end
+
+      # Command for writing the useradd default values
+      USERADD = "/usr/sbin/useradd".freeze
+      private_constant :USERADD
+
+      # Mapping between {UseraddConfig} attributes and useradd arguments
+      USERADD_ATTRS = {
+        group:             "--gid",
+        home:              "--base-dir",
+        shell:             "--shell",
+        expiration:        "--expiredate",
+        inactivity_period: "--inactive"
+      }.freeze
+      private_constant :USERADD_ATTRS
+
+      # Writes the attributes that are handled via "useradd -D"
+      #
+      # @param issues [Y2Issues::List]
+      def write_useradd(issues)
+        return unless write_useradd?
+
+        # Instead of modifying directly the /etc/default/useradd file using the agent
+        # etc.default.useradd, we rely on "useradd -D". That should be more future-proof because:
+        #   - It should keep working if some of the parameters is moved to another file
+        #   - It will report an issue if we write a value that useradd considers to be wrong
+        #
+        # useradd allows to specify several default values at one shot like this:
+        #   useradd -D --base-dir /people --gid users --shell /bin/zsh
+        # But that works on a all-or-nothing fashion, ie. if one of the values is wrong (eg. the
+        # group "users" does not exist) none of the values is written
+        #
+        # To reduce the impact of a wrong value, let's change them one by one
+        USERADD_ATTRS.each do |attr, arg_name|
+          configure_useradd_attr(attr, arg_name, issues)
+        end
+      end
+
+      # Whether executing "useradd -D" is really needed
+      #
+      # @return [Boolean]
+      def write_useradd?
+        return false if useradd_config.nil?
+        return true if initial_useradd_config.nil?
+
+        USERADD_ATTRS.keys.any? { |attr| value(attr) != initial_value(attr) }
+      end
+
+      # Writes the attributes that are handled via login.defs
+      def write_shadow_config
+        return unless write_shadow_config?
+
+        Yast::ShadowConfig.set(:umask, value(:umask))
+        Yast::ShadowConfig.write
+      end
+
+      # Whether writing to login.defs is really needed
+      #
+      # @return [Boolean]
+      def write_shadow_config?
+        return false if value(:umask).nil?
+
+        value(:umask) != initial_value(:umask)
+      end
+
+      # @see #write_useradd
+      #
+      # @param attr [Symbol]
+      # @param arg [String]
+      # @param issues [Y2Issues::List]
+      def configure_useradd_attr(attr, arg, issues)
+        value = value(attr)
+        return if value.nil?
+
+        Yast::Execute.on_target!(USERADD, "-D", arg, value)
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          # TRANSLATORS: %s is the name of one of the useradd default values, like 'HOME' or 'GROUP'
+          format(_("Something went wrong writing the useradd default for '%s'"), arg)
```

Comment:
I think this issues message is something we need to address. It is quite useless now. You just know that something went wrong, but not if it is written or not, what is error message, etc.

Reply:
Giving a more accurate message would likely imply parsing the stderr of "useradd -D" and similar stuff that I consider overkill taking into consideration the details can be looked up in the YaST logs. But I'm open to suggestions here.

Reply:
ok, I already discuss it with imo that it would be nice if there is something like details where we can just pass untranslated stderr.